Core/clim-core/presentations/presentation-types.lisp

```diff
@@ -57,6 +57,32 @@
   (:documentation "The specializer to use for this type in a presentation
 method lambda list"))
 
+(defmethod ptype-specializer ((type symbol))
+  (if-let ((ptype (or (find-presentation-type type nil)
+                      (find-class type nil))))
+    (ptype-specializer ptype)
+    ;; Assume it's a forward referenced CLOS class.
+    type))
+
+(defmethod ptype-specializer ((type standard-class))
+  (class-name type))
+
+(defmethod ptype-specializer ((type cons))
+  (if (typep type '(cons (eql eql)
+                    (cons t null)))
+      type
+      (error "~s is not a valid presentation method specializer." type)))
+
+(defmacro destructure-type-arg
+    ((type-var type-spec type-name) type-arg &body body)
+  `(destructuring-bind (,type-var ,type-spec) ,type-arg
+     (let ((,type-var ,type-var)
+           (,type-spec ,type-spec)
```

Comment:
to avoid style warnings, i.e

```
(destructure-arg (foo bar qux) x
  (declare (ignore bar qux))
  …)
```

Reply:
I don't understand:
* `type-spec` is always used in the computation of the value for `type-name`
* If `body` does not use `type-var`, the new binding will cause a warning instead

Is there a reason not to `(declare (ignorable ,type-var ,type-name))`?

Reply:
In here we have:

```
(defmacro define-subtypep-method (&rest args)
  (let ((gf (find-presentation-gf 'presentation-subtypep)))
    (multiple-value-bind (qualifiers lambda-list decls body)
        (parse-method-body args)
      (let ((type-arg (nth (1- (type-arg-position gf)) lambda-list)))

        (unless (consp type-arg)
          (error "Type argument in presentation method must be specialized"))
        (unless (eq (car type-arg)  'type)
          (error "Type argument mismatch with presentation generic function
 definition"))
        (destructure-type-arg (type-var type-spec type-name) type-arg
          (declare (ignore type-spec))
          (let ((method-ll `((,(type-key-arg gf)
                              (eql (prototype-or-error ',type-name)))
                             ,@(copy-list lambda-list))))
            (setf (nth (type-arg-position gf) method-ll) type-var)
            `(defmethod %presentation-subtypep ,@qualifiers ,method-ll
               (declare (ignorable ,(type-key-arg gf))
                        ,@(cdr decls))
               (block presentation-subtypep
                 ,@body))))))))
```

notice, that the type-spec is indeed ignored. adding declare ignorable is fine too, I don't see any meaningful difference.

Reply:
excepta that this declaration (in the code) would  be invalid, because it would be misplaced. so it would be at least surprising to the programmer that they can't declare ignored variable as ignored.

Reply:
I see. I misunderstood what you meant by "to avoid style warnings, i.e" because I read `destructure-arg` as `destructuring-bind` in the following snippet.